core/new-gui/src/app/workspace/service/workflow-result-export/workflow-result-export.service.ts

```diff
@@ -0,0 +1,66 @@
+import { Injectable } from '@angular/core';
+import { environment } from '../../../../environments/environment';
+import { WorkflowWebsocketService } from '../workflow-websocket/workflow-websocket.service';
+import { WorkflowActionService } from '../workflow-graph/model/workflow-action.service';
+import { Observable } from 'rxjs/Observable';
+import { ResultDownloadResponse } from '../../types/workflow-websocket.interface';
+import { NotificationService } from '../../../common/service/notification/notification.service';
+import { ExecuteWorkflowService } from '../execute-workflow/execute-workflow.service';
+import { ExecutionState } from '../../types/execute-workflow.interface';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class WorkflowResultExportService {
+  private hasResultToExport: boolean = false;
+
+  constructor(
+    private workflowWebsocketService: WorkflowWebsocketService,
+    private workflowActionService: WorkflowActionService,
+    private notificationService: NotificationService,
+    private executeWorkflowService: ExecuteWorkflowService
+  ) {
+    this.workflowWebsocketService.subscribeToEvent('ResultDownloadResponse').subscribe((response: ResultDownloadResponse) => {
```

Comment:
I do think these logic can be put inside a component instead of a service.

Reply:
There is no corresponding component for export/download. The only place might be the navigation component. but I think that's not suitable either.

This can be put here as it is part of the service, it handles the responses from the WebSocket, and forward to Notification Component for displaying messages.